aws/resource_aws_prometheus_workspace_test.go

```diff
@@ -0,0 +1,140 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAMPWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAWSAMPWorkspaceConfigWithAlias(workspaceAlias),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAWSAMPWorkspaceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "alias", workspaceAlias),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAWSAMPWorkspaceConfigWithoutAlias(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "alias", ""),
+				),
+			},
+			{
+				Config: testAWSAMPWorkspaceConfigWithAlias(workspaceAlias),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAWSAMPWorkspaceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "alias", workspaceAlias),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSAMPWorkspace_disappears(t *testing.T) {
+	resourceName := "aws_prometheus_workspace.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAMPWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAWSAMPWorkspaceConfigWithoutAlias(),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAWSAMPWorkspaceExists(resourceName),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsPrometheusWorkspace(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testCheckAWSAMPWorkspaceExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No AMP Workspace ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).prometheusserviceconn
+
+		req := &prometheusservice.DescribeWorkspaceInput{
+			WorkspaceId: aws.String(rs.Primary.ID),
+		}
+		describe, err := conn.DescribeWorkspace(req)
+		if err != nil {
+			return err
+		}
+		if describe == nil {
+			return fmt.Errorf("Got nil account ?!")
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSAMPWorkspaceDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).prometheusserviceconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_prometheus_workspace" {
+			continue
+		}
+
+		res, err := conn.DescribeWorkspace(&prometheusservice.DescribeWorkspaceInput{
+			WorkspaceId: aws.String(rs.Primary.ID),
+		})
+		if err == nil {
+			if aws.StringValue(res.Workspace.Status.StatusCode) != "DELETING" {
+				return fmt.Errorf("AMP workspace still exists")
+			}
+		}
+
+		// Verify the error is what we want
+		if isAWSErr(err, prometheusservice.ErrCodeResourceNotFoundException, "") {
+			return nil
+		}
+
+		return err
```

Comment:
We should be able to remove the `DELETING` check now here, to verify that indeed the Prometheus WorkSpace was deleted. 👍 

```suggestion
		if isAWSErr(err, prometheusservice.ErrCodeResourceNotFoundException, "") {
			continue
		}

		if err != nil {
			return fmt.Errorf("error reading Prometheus WorkSpace (%s): %w", rs.Primary.ID, err)
		}
```